### PR TITLE
BaSP-TG: Projects Edit / Add Employee / Get by filters

### PR DESCRIPTION
**Pull request by JulianFloresDev**
Create Pull Request to include Projects Edit / Add Employee & Get by Filters component.

### CODE REVIEW

**Comment by NicolasGulam** [comment]
Juli fijate que no respetaste esto: El nombre de las PR deben respetar el formato BaSP-TG: Short description

File: src/resources/projects.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+/* Get projects by argument
+? On body must have almost 1 params between:
+? "id"
+? "name"
+? "clientName"
+? "startDate"
+? "endDate"
+*/
+router.get('/getByArgument', (req, res, next) => {
```
Comment (ignacioaristo) [inline src/resources/projects.js:156]: Same for this next

File: src/resources/projects.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+/* eslint-disable import/no-import-module-exports */
+import express from 'express';
+
+import fs from 'fs';
+
+const projectsList = require('../data/projects.json');
+
+const router = express.Router();
+
+const findProjectOnReq = (request) => {
```
Comment (ignacioaristo) [inline src/resources/projects.js:12]: Nice function.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]
All the messages in github must be in English.

File: src/index.js [hunk 3]
```diff
@@ -13,6 +13,10 @@ app.get('/', async (req, res) => {
   res.send('Hello World!');
 });
 
+//! Julian Flores & Diaz Camila workspace:
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/index.js]: Change Done!

File: src/index.js [hunk 4]
```diff
@@ -13,6 +13,10 @@ app.get('/', async (req, res) => {
...
+app.use('/projects', require('./resources/projects'));
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/index.js]: Change Done!

File: .vscode/settings.json [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
-  "editor.tabSize": 4,
+  "editor.tabSize": 2,
   "editor.bracketPairColorization.enabled": true,
   "editor.guides.bracketPairs": "active",
   "editor.guides.bracketPairsHorizontal": "active",
   "editor.codeActionsOnSave": {
     "source.fixAll.eslint": true
   },
   "eslint.format.enable": true,
+  "editor.fontSize": 15,
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply .vscode/settings.json:12]: Change Done!

File: src/resources/projects.js [hunk 6]
```diff
@@ -0,0 +1,167 @@
+/* eslint-disable max-len */
+/* eslint-disable no-prototype-builtins */
+/* eslint-disable import/no-import-module-exports */
+import express from 'express';
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/resources/projects.js]: Change Done!

File: src/resources/projects.js [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+  });
+  return employeeToDelete;
+};
+
+/*
+!Edit Project
+? On route must have ID number after /
+? On body must have Property to edit with new value.
+*/
+router.put('/edit/project/:id', (req, res) => {
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/resources/projects.js]: Change Done!

File: src/resources/projects.js [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+      } else {
+        res.send('Employee Delete Successfully')
+          .json(filterProjects);
+      }
+    });
+  }
+});
+
+//! Get projects
+router.get('/getAll', (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/resources/projects.js]: Change Done!

[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/resources/projects.js]: Change Done!

File: src/resources/projects.js (hunk 1 above)
Comment (JulianFloresDev) [reply src/resources/projects.js]: Change Done!

File: src/resources/projects.js (hunk 2 above)
Comment (JulianFloresDev) [reply src/resources/projects.js:10]: Thanks!

File: .vscode/settings.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (ignacioaristo) [reply .vscode/settings.json:12]: Juli, we need to fix something. Remove the file from the git ignore, delete these changes, push again. And then add those lines of code and add the file into de ignore. If you have any doubt MD me.

[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply .vscode/settings.json]: Changes done!!

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**